src/event.js

```diff
@@ -573,7 +574,7 @@ jQuery.Event.prototype = {
 
 		this.isPropagationStopped = returnTrue;
 
-		if ( e ) {
+		if ( e && this.executeOriginalPropagation ) {
```

Comment:
I see. But rereading the [`simulate` comments](https://github.com/jquery/jquery/blob/305f193aa57014dc7d8fa0739a3fefd47166cd44/src/event/trigger.js#L166-L167), it seems more like our folly than Mozilla's. I'd actually advocate for dropping the "Support" documentation here altogether in favor uniform simulated event treatment:
```js
stopPropagation: function() {
	var e = this.originalEvent;

	this.isPropagationStopped = returnTrue;

	// Stop propagation of original non-donor events
	if ( !this.isSimulated && e ) {
		e.stopPropagation();
	}
}
```
(and likewise for `stopImmediatePropagation`).